document/execution/numerics.rst

```diff
@@ -1,54 +1,1472 @@
+* Else if :math:`z_1` is negative infinity, then return :math:`1`.
+
+* Else if :math:`z_2` is positive infinity, then return :math:`1`.
+
+* Else if both :math:`z_1` and :math:`z_2` are zeroes, then return :math:`1`.
+
+* Else if :math:`z_1` is smaller than or equal to :math:`z_2`, then return :math:`1`.
+
+* Else return :math:`0`.
+
+.. math::
+   \begin{array}{@{}lcll}
+   \fle_N(\pm \NAN(n), z_2) &=& 0 \\
+   \fle_N(z_1, \pm \NAN(n)) &=& 0 \\
+   \fle_N(z, z) &=& 1 \\
+   \fle_N(- \infty, z_2) &=& 1 \\
+   \fle_N(z_1, + \infty) &=& 1 \\
+   \fle_N(\pm 0, \mp 0) &=& 1 \\
+   \fle_N(z_1, z_2) &=& \bool(z_1 \leq z_2) \\
+   \end{array}
+
+
+.. _op-fge:
+
+:math:`\fge_N(z_1, z_2)`
+........................
+
+* If either :math:`z_1` or :math:`z_2` is a NaN, then return :math:`0`.
+
+* Else if :math:`z_1` and :math:`z_2` are the same value, then return :math:`1`.
+
+* Else if :math:`z_1` is positive infinity, then return :math:`1`.
+
+* Else if :math:`z_2` is negative infinity, then return :math:`1`.
+
+* Else if both :math:`z_1` and :math:`z_2` are zeroes, then return :math:`1`.
+
+* Else if :math:`z_1` is smaller than or equal to :math:`z_2`, then return :math:`1`.
+
+* Else return :math:`0`.
+
+.. math::
+   \begin{array}{@{}lcll}
+   \fge_N(\pm \NAN(n), z_2) &=& 0 \\
+   \fge_N(z_1, \pm \NAN(n)) &=& 0 \\
+   \fge_N(z, z) &=& 1 \\
+   \fge_N(+ \infty, z_2) &=& 1 \\
+   \fge_N(z_1, - \infty) &=& 1 \\
+   \fge_N(\pm 0, \mp 0) &=& 1 \\
+   \fge_N(z_1, z_2) &=& \bool(z_1 \geq z_2) \\
+   \end{array}
+
+
+Conversions
+~~~~~~~~~~~
+
+.. _op-extendu:
+
+:math:`\extendu_{M,N}(i)`
+.........................
+
+* Return :math:`i`.
+
+.. math::
+   \begin{array}{lll@{\qquad}l}
+   \extendu_{M,N}(i) &=& i \\
+   \end{array}
+
+.. note::
+   In the abstract syntax, unsigned extension just reinterprets the same value.
+
+
+.. _op-extends:
+
+:math:`\extends_{M,N}(i)`
+.........................
+
+* Let :math:`j` be the signed interpretation of :math:`i` of size :math:`M`.
+
+* Return the 2's complement of :math:`j` relative to size :math:`N`.
+
+.. math::
+   \begin{array}{lll@{\qquad}l}
+   \extends_{M,N}(i) &=& \signed_N^{-1}(\signed_M(i)) \\
+   \end{array}
+
+
+.. _op-wrap:
+
+:math:`\wrap_{M,N}(i)`
+......................
+
+* Return :math:`i` modulo :math:`N`.
+
+.. math::
+   \begin{array}{lll@{\qquad}l}
+   \wrap_{M,N}(i) &=& i \mod 2^N \\
+   \end{array}
+
+
+.. _op-truncu:
+
+:math:`\truncu_{M,N}(z)`
+........................
+
+* If :math:`z` is a NaN, then the result is undefined. 
+
+* Else if :math:`z` is an infinity, then the result is undefined. 
+
+* Else if :math:`z` is a number and :math:`\trunc(z)` is a value within range of the target type, then return that value.
+
+* Else the result is undefined.
+
+.. math::
+   \begin{array}{lll@{\qquad}l}
+   \truncu_{M,N}(\pm \NAN(n)) &=& \{\} \\
+   \truncu_{M,N}(\pm \infty) &=& \{\} \\
+   \truncu_{M,N}(\pm 0) &=& 0 \\
+   \truncu_{M,N}(\pm q) &=& \trunc(\pm q) & (0 \leq \trunc(\pm q) < 2^N) \\
```

Comment:
The valid range is (-1 < \trunc(\pm q) < 2^N) and not (0 \leq \trunc(\pm q) < 2^N)